clientgui/DlgAdvPreferencesBase.cpp

```diff
@@ -250,233 +249,296 @@ wxPanel* CDlgAdvPreferencesBase::createProcessorTab(wxNotebook* notebook)
+    m_chkMaxLoad = new wxCheckBox(
+        box, ID_CHKMAXLOAD,
+        wxString(_("Suspend when non-BOINC CPU usage is above")),
         wxDefaultPosition, wxDefaultSize, 0
     );
+    m_txtMaxLoad = new wxTextCtrl(
+        box, ID_TXTMAXLOAD, wxEmptyString, wxDefaultPosition, getTextCtrlSize(wxT("100.00")), wxTE_RIGHT
+    );
+    addNewRowToSizer(box_sizer,
+        wxString (_("Suspend computing when your computer is busy running other programs.")),
+        m_chkMaxLoad, m_txtMaxLoad,
+        new wxStaticText(box, ID_DEFAULT, wxT("%"), wxDefaultPosition, wxDefaultSize, 0)
+    );
 
-    addNewRowToSizer(suspendComputingBoxSizer, ProcIdleForTT, staticText24, m_txtProcIdleFor, staticText25);
+    // max memory while in use
+    //
+    wxString MemoryMaxInUseTT = wxEmptyString;
+    MemoryMaxInUseTT.Printf(_("Limit the memory used by %s when you're using the computer."), pSkinAdvanced->GetApplicationShortName().c_str());
+    wxStaticText* staticText50 = new wxStaticText(box, ID_DEFAULT, _("Use at most"), wxDefaultPosition, wxDefaultSize, 0 );
+    textCtrlSize = getTextCtrlSize(wxT("100.00"));
+    m_txtMemoryMaxInUse = new wxTextCtrl( box, ID_TXTMEMORYMAXINUSE, wxEmptyString, wxDefaultPosition, textCtrlSize, wxTE_RIGHT );
+    /*xgettext:no-c-format*/
+    wxStaticText* staticText51 = new wxStaticText( box, ID_DEFAULT, _("% of memory"), wxDefaultPosition, wxDefaultSize, 0 );
+    addNewRowToSizer(box_sizer, MemoryMaxInUseTT, staticText50, m_txtMemoryMaxInUse, staticText51);
 
-    // suspend after max idle time
-    wxString NoRecentInputTT(_("This allows some computers to enter low-power mode when not in use."));
+    processorTabSizer->AddSpacer( STATICBOXVERTICALSPACER );
+    processorTabSizer->Add(box_sizer, 0, wxLEFT | wxRIGHT | wxEXPAND, STATICBOXBORDERSIZE );
+
+    // ------------ Not-in-use box --------------
+    //
+
+    box = new wxStaticBox(processorTab, -1, _("When computer is not in use") );
+    box_sizer = new wxStaticBoxSizer(box, wxVERTICAL);
+    makeStaticBoxLabelItalic(box);
+
+    // max # CPUs
+    //
+    /*xgettext:no-c-format*/
+    wxString MaxCPUPctTTniu(_("Keep some CPUs free for other applications. Example: 75% means use 6 cores on an 8-core CPU."));
+    wxStaticText* staticText20niu = new wxStaticText(
+        box, ID_DEFAULT, _("Use at most"), wxDefaultPosition, wxDefaultSize, 0
+    );
+    m_txtProcUseProcessorsNotInUse = new wxTextCtrl(box, ID_TXTPROCUSEPROCESSORSNOTINUSE, wxEmptyString, wxDefaultPosition, textCtrlSize, wxTE_RIGHT);
+    /*xgettext:no-c-format*/
+    wxStaticText* staticText21niu = new wxStaticText(box, ID_DEFAULT, _("% of the CPUs"), wxDefaultPosition, wxDefaultSize, 0);
+    addNewRowToSizer(box_sizer, MaxCPUPctTTniu, staticText20niu, m_txtProcUseProcessorsNotInUse, staticText21niu);
 
-    wxStaticText* staticText27 = new wxStaticText(
-        suspendComputingStaticBox, ID_DEFAULT,
-        _("Suspend when no mouse/keyboard input in last"),
+    // CPU throttling
+    //
```

Comment:
Precede line 375 with ` /*xgettext:no-c-format*/`

Reply:
@davidpanderson You have not fixed this.  For an explanation why this is needed, see [here](https://stackoverflow.com/questions/70490036/xgettext-incorrectly-infers-c-format-in-cpp-files) and [here](https://www.mit.edu/afs.new/athena/astaff/source/src-9.4/third/gettext2/gettext-tools/doc/gettext_3.html#SEC18).

(Of course the `/*xgettext:no-c-format*/` directive must now precede line 376 in the latest version of this file.)